### PR TITLE
Fixed 2 bugs in LocalChanges module

### PR DESCRIPTION
**Pull request by grondinjc**
51 tests to back this up :+1:

### CODE REVIEW

**Comment by grondinjc** [comment]
@vphilippon  Review

**Comment by grondinjc** [comment]
or @maym2104

**Comment by vphilippon** [comment]
@grondinjc I'll get on this soon<sup>tm</sup>.

File: src/server/ideController.py [hunk 1]
```diff
@@ -371,17 +384,23 @@ def _save_callback(self, filename, changes, version, users):
       }
     """
 
-    temp_message = ("Temp msg from controller until "
-                    "c++ module will return applied modifications")
-    changes = [dict(type=self.CHANGE_ADD_TYPE, pos=0, content=temp_message)]
+    # Dirty hack to know type
```
Comment (vphilippon) [inline src/server/ideController.py:387]: @grondinjc Do you have another idea to remove this hack later? I feel like we should take care of this now or really soon.

Comment (grondinjc) [inline src/server/ideController.py:387]: The change object return by c++ should contain the flag type to solve this

Comment (maym2104) [inline src/server/ideController.py:387]: Will add it

File: src/static/js/modules/requestHandler.js [hunk 2]
```diff
@@ -51,9 +51,10 @@ RequestHandler.prototype.close = function() {
     this._socket.close();
   }
 };
-RequestHandler.prototype._send = function(type, url, requestData, successCallback, errorCallback) {
+RequestHandler.prototype._send = function(type, url, requestData, successCallback, errorCallback, async) {
   successCallback = successCallback || this.EMPTY_CALLBACK;
   errorCallback = errorCallback || this.EMPTY_CALLBACK;
+  async = async == undefined ? true : false;
```
Comment (vphilippon) [inline src/static/js/modules/requestHandler.js]: If the `async` is `true`, it won't be `undefined`, and then set to false. Your 'else' value should be `async`
Ex: `async = async == undefined ? true : async;`

Comment (vphilippon) [inline src/static/js/modules/requestHandler.js]: Also, `async` is unused. I guess it's missing in the ajax parameters

File: src/server/ideController.py (hunk 1 above)
Comment (vphilippon) [inline src/server/ideController.py]: Ok. Solved in a later PR.

**Pull request merged**